raiden_contracts/tests/deprecation_switch_testnet.py

```diff
@@ -0,0 +1,238 @@
+import click
+from logging import getLogger
+from eth_utils import encode_hex
+
+from raiden_libs.private_contract import PrivateContract
+
+from raiden_contracts.utils.utils import check_succesful_tx
+from raiden_contracts.deploy.__main__ import (
+    setup_ctx,
+    deploy_raiden_contracts,
+    deploy_token_contract,
+    register_token_network,
+)
+from raiden_contracts.constants import (
+    CONTRACT_CUSTOM_TOKEN,
+    CONTRACT_TOKEN_NETWORK_REGISTRY,
+    CONTRACT_TOKEN_NETWORK,
+    MAX_ETH_CHANNEL_PARTICIPANT,
+    DEPLOY_SETTLE_TIMEOUT_MIN,
+)
+
+
+log = getLogger(__name__)
+
+
+@click.command()
+@click.option(
+    '--private-key',
+    required=True,
+    help='Path to a private key store.',
+)
+@click.option(
+    '--rpc-provider',
+    default='http://127.0.0.1:8545',
+    help='Address of the Ethereum RPC provider',
+)
+@click.option(
+    '--wait',
+    default=300,
+    help='Max tx wait time in s.',
+)
+@click.option(
+    '--gas-price',
+    default=5,
+    type=int,
+    help='Gas price to use in gwei',
+)
+@click.option(
+    '--gas-limit',
+    default=5_500_000,
+)
+@click.pass_context
+def deprecation_test(
+    ctx,
+    private_key,
+    rpc_provider,
+    wait,
+    gas_price,
+    gas_limit,
+):
+    setup_ctx(ctx, private_key, rpc_provider, wait, gas_price, gas_limit)
+    deployer = ctx.obj['deployer']
+
+    # We deploy the Raiden Network contracts and register a token network
+    token_amount = MAX_ETH_CHANNEL_PARTICIPANT * 6
+    (
+        token_network_registry,
+        token_network,
+        token_contract,
+    ) = deprecation_test_setup(deployer, token_amount)
+
+    log.info('Checking that channels can be opened and deposits can be made.')
+
+    # Check that we can open channels and deposit on behalf of A and B
+    # Some arbitrary Ethereum addresses
+    A = '0x6AA63296FA94975017244769F00F0c64DB7d7115'
+    B = '0xc9a4fad99B6d7D3e48D18d2585470cd8f27FA61e'
+    channel_identifier = open_and_deposit(A, B, token_network, deployer)
+    log.info('Seding transaction to activate the deprecation switch.')
+
+    # Activate deprecation switch
+    txhash = token_network.functions.deprecate().transact(
+        deployer.transaction,
+        private_key=deployer.private_key,
+    )
+    log.debug(f'Deprecation txHash={encode_hex(txhash)}')
+    check_succesful_tx(deployer.web3, txhash, deployer.wait)
+    assert token_network.functions.safety_deprecation_switch().call() is True
+
+    log.info('Checking that channels cannot be opened anymore and no more deposits are allowed.')
+
+    # Check that we cannot open more channels or deposit
+    C = '0x5a23cedB607684118ccf7906dF3e24Efd2964719'
+    D = '0x3827B9cDc68f061aa614F1b97E23664ef3b9220A'
+    open_and_deposit(C, D, token_network, deployer, channel_identifier)
```

Comment:
Actually, in `open_and_deposit ` the `openChannel` or `setTotalDeposit` transactions will fail with `ValueError: {'code': -32000, 'message': 'gas required exceeds allowance or always failing transaction'}`. I catch the error for each tx and log it:
```
except ValueError as ex:
        log.info(f'Cannot open a new channel {ex}')
```

```
except ValueError as ex:
        log.info(f'Cannot deposit more tokens in channel={channel_identifier}, {ex}')
```

Reply:
Yes I saw this.  But that's not a test. You should assert on them failing. Not just log them.

If somehow the deprecation switch breaks and channel creation works after it's activated it's better to have a test fail rather than having a human have to manually check the log and make a decision if it's fine or not. Not to mention it would not work in Travis or any other automated CI.

I know it's not made to run as a pytest but at least you can make it fail and return an error code if something does not work as expected.

Reply:
Right! Added txn status assertions in https://github.com/raiden-network/raiden-contracts/pull/338/commits/ef05df2cbbee77cc3c08e011de194b94553d0b0a
The `assert`s fail if the switch is not triggered.